examples/WORKSPACE.bazel

```diff
@@ -79,15 +79,20 @@ rust_repository_set(
 # Examples dependencies
 ###############################################################################
 
-http_archive(
-    name = "build_bazel_rules_nodejs",
-    sha256 = "c78216f5be5d451a42275b0b7dc809fb9347e2b04a68f68bad620a2b01f5c774",
-    urls = ["https://github.com/bazelbuild/rules_nodejs/releases/download/5.5.2/rules_nodejs-5.5.2.tar.gz"],
+load("@aspect_rules_js//js:repositories.bzl", "rules_js_dependencies")
+
+rules_js_dependencies()
+
+load("@rules_nodejs//nodejs:repositories.bzl", "DEFAULT_NODE_VERSION", "nodejs_register_toolchains")
+
+nodejs_register_toolchains(
+    name = "nodejs",
+    node_version = DEFAULT_NODE_VERSION,
 )
 
-load("@build_bazel_rules_nodejs//:index.bzl", "node_repositories")
+load("@bazel_features//:deps.bzl", "bazel_features_deps")
```

Comment:
What is this? The repo seems to come out of nowhere. Is it required to use `aspect_rules_js`?

Reply:
tl;dr This is a bit of a wart, yes.

The package we load the JsInfo provider from depends on that repo to define a config setting: https://github.com/aspect-build/rules_js/blob/63862adcfb565b0a53fb45c60dded0f8a7893c49/js/BUILD.bazel#L3C1-L15 This seems designed to support multiple Bazel versions so hopefully it can go away at some point. That repo also has a small usability trick - it automatically calls `bazel_features_deps` if you use `npm_translate_lock`, and since the vast majority of JS projects use npm packages, they don't have to do anything special here.

I do agree it's a bit weird to have to pull this in just to use the provider, @alexeagle perhaps you can consider rearranging the layout to make the providers more standalone?

Reply:
@UebelAndre What do you think?

Reply:
Hm, yeah I see why this is undesirable. We had to make those part of the public API in https://github.com/aspect-build/rules_js/pull/1187 - but maybe rules_rust can load() the providers without having to load that package? providers.bzl is in its own file for this reason.

Reply:
The problem is the `//docs:docs_deps` rule in this repo. The bzl_library requires depending on `"@aspect_rules_js//js:providers"` to satisfy the stardoc rules/lint, which means anything loaded in `//js:BUILD` is a dependency. I think if you moved the providers.bzl to its own package (say `//js/providers/defs.bzl`) it would work.

Reply:
The bzl_library here can just depend on the bzl file directly in srcs to avoid that

Reply:
Not sure I'm following the suggestion, do you mean like so?
```
bzl_library(
    name = "docs_deps",
    srcs = [
        "@aspect_rules_js//js:providers.bzl",
        "@bazel_tools//tools:bzl_srcs",
        "@com_google_protobuf//:bzl_srcs",
    ],
   deps = [...]
)
```

That hits visibility issues: https://github.com/aspect-build/rules_js/blob/main/js/BUILD.bazel#L6-L9
Even ignoring visibility, it will have a problem with missing the js_info bzl
```
com.google.devtools.build.skydoc.SkydocMain$StarlarkEvaluationException: File /private/var/tmp/_bazel_zbarsky/e27223f43abb18db045469720c4d6593/sandbox/darwin-sandbox/28/execroot/rules_rust_docs/bazel-out/darwin-opt-exec-2B5CBBC6/bin/rust_repositories_md_stardoc.runfiles/aspect_rules_js/js/providers.bzl imported '//js/private:js_info.bzl', yet /private/var/tmp/_bazel_zbarsky/e27223f43abb18db045469720c4d6593/sandbox/darwin-sandbox/28/execroot/rules_rust_docs/bazel-out/darwin-opt-exec-2B5CBBC6/bin/rust_repositories_md_stardoc.runfiles/aspect_rules_js/js/private/js_info.bzl was not found.
```

Reply:
@UebelAndre ping? What can I do to get this PR moving?

Reply:
I’ll try to make time this weekend to do another pass. Sorry for the delay!